hummingbot/connector/exchange/bittrex/bittrex_exchange.py

```diff
@@ -0,0 +1,400 @@
+
+    async def _request_order_status(self, tracked_order: InFlightOrder) -> OrderUpdate:
+        exchange_order_id = await tracked_order.get_exchange_order_id()
+        params = {
+            "orderId": exchange_order_id
+        }
+        order_update = await self._api_get(
+            path_url=CONSTANTS.ORDER_DETAIL_URL.format(exchange_order_id),
+            params=params,
+            is_auth_required=True,
+            limit_id=CONSTANTS.ORDER_DETAIL_LIMIT_ID
+        )
+        new_state = self._get_order_status(order_update)
+        update_time = _get_timestamp(order_update["updatedAt"])
+        update = OrderUpdate(
+            client_order_id=tracked_order.client_order_id,
+            exchange_order_id=exchange_order_id,
+            trading_pair=tracked_order.trading_pair,
+            update_timestamp=update_time,
+            new_state=new_state,
+        )
+        return update
+
+    async def _user_stream_event_listener(self):
+        async for stream_message in self._iter_user_event_queue():
+            try:
+                content = stream_message.get("delta") or stream_message.get("deltas")
+                if isinstance(content, List):
+                    safe_ensure_future(self._process_execution_event(content))
+                elif "marketSymbol" not in content:
+                    asset_name = content["currencySymbol"]
+                    total_balance = Decimal(content["total"])
+                    available_balance = Decimal(content["available"])
+                    self._account_available_balances[asset_name] = available_balance
+                    self._account_balances[asset_name] = total_balance
+                else:
+                    safe_ensure_future(self._process_order_update_event(content))
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().error("Unexpected error in user stream listener loop.", exc_info=True)
+
+    async def _process_order_update_event(self, msg: Dict[str, Any]):
+        order_id = msg["id"]
+        update_time = _get_timestamp(msg["updatedAt"])
+        order_state = self._get_order_status(msg)
+        tracked_order = None
+        for order in self._order_tracker.all_updatable_orders.values():
+            if order.exchange_order_id is None:
+                continue
+            exchange_order_id = await order.get_exchange_order_id()
+            if exchange_order_id == order_id:
+                tracked_order = order
+                break
+        if tracked_order is not None:
+            order_update = OrderUpdate(
+                trading_pair=tracked_order.trading_pair,
+                update_timestamp=update_time,
+                new_state=order_state,
+                client_order_id=tracked_order.client_order_id,
+                exchange_order_id=order_id,
+            )
+            self._order_tracker.process_order_update(order_update=order_update)
```

Comment:
Each order update event includes the `clientOrderId` (according to https://bittrex.github.io/api/v3#method-Order). We should use that information to get the order we want to update instead of cycling through all the orders

Reply:
Implemented a separate private method in superclass for generating clientorderid, which is overriden in connector class. We can send and get the clientorderid from the exchange now.